### ISSUE TITLE
The error of make

### ISSUE THREAD

**Issue by lqs661** [ukaea/ReMKiT1D#5]
When I executed 'make', the errors followed as:
[100%] Linking Fortran executable ReMKiT1D
/usr/bin/ld: ../../modules/solver_support/libsolver_support.a(petsc_controller_procedures.f90.o): in function `__petsc_controller_class_MOD_createpetscobjs':
petsc_controller_procedures.f90:(.text+0xddf): undefined reference to `pchypresettype_'
collect2: error: ld returned 1 exit status
make[2]: *** [src/executables/ReMKiT1D/CMakeFiles/ReMKiT1D.dir/build.make:148: src/executables/ReMKiT1D/ReMKiT1D] Error 1
make[1]: *** [CMakeFiles/Makefile2:1966: src/executables/ReMKiT1D/CMakeFiles/ReMKiT1D.dir/all] Error 2
make: *** [Makefile:149: all] Error 2

**Comment by lqs661** [comment]
The step of building ReMKiT1D

**Comment by SMijin** [comment]
Hi there, thanks for raising this issue. This looks like your PETSc installation might not have hypre support. Could you please check whether it's been configured with hypre? 

Could you also please share the cmake output before you called make?

**Comment by SMijin** [comment]
Hmm, this looks like some PETSc issue. I'd double check that your compilers/paths are all ok, but other than that I'm not sure what could be the problem, especially since I don't have the full output of PETSc's configure command (where it might have given you some warnings and continued on its merry way).

You mention you git cloned hyper instead of letting PETSc download it. Could that be the source of the issue?

**Comment by lqs661** [comment]
> Hmm, this looks like some PETSc issue. I'd double check that your compilers/paths are all ok, but other than that I'm not sure what could be the problem, especially since I don't have the full output of PETSc's configure command (where it might have given you some warnings and continued on its merry way).
> 
> You mention you git cloned hyper instead of letting PETSc download it. Could that be the source of the issue?

Yes, I think the hypre caused this issue. In China, because of the connection problem, I must git clone hypre using proxy by myself instead of letting PETSc download. When I downloaded hypre I didn't notice the version of hypre was latest, which may be uncompatible with PETSc 3.17. I changed the hypre from 2.29.0 to 2.24.0, and the problem was resolved. When I executed 'make', it could build target ReMKiT1D. 
However, when I executed 'make test', it showed "No rule to make target 'test'". I don't know whether this impact the program execution.

**Comment by SMijin** [comment]
Have you installed pFUnit and set the proper environment variables? CMake won't build tests unless you have pFUnit installed. 

It shouldn't impact the actual execution of the code, but it's always good to test when you compile on a new system.
